src/particle/pdgid/pdgid.py

```diff
@@ -63,8 +63,47 @@ def info(self):
             val += "{item:14} {value}\n".format(item=item, value=getattr(self, item))
         return val
 
-
-# Decorate the PDGID class with all relevant functions defined in the pdgid.functions module
+    A = property(_functions.A)
+    J = property(_functions.J)
+    L = property(_functions.L)
+    S = property(_functions.S)
+    Z = property(_functions.Z)
+    abspid = property(_functions.abspid)
+    charge = property(_functions.charge)
+    has_bottom = property(_functions.has_bottom)
+    has_charm = property(_functions.has_charm)
+    has_down = property(_functions.has_down)
+    has_fundamental_anti = property(_functions.has_fundamental_anti)
+    has_strange = property(_functions.has_strange)
+    has_top = property(_functions.has_top)
+    has_up = property(_functions.has_up)
+    is_Qball = property(_functions.is_Qball)
+    is_Rhadron = property(_functions.is_Rhadron)
+    is_SUSY = property(_functions.is_SUSY)
+    is_baryon = property(_functions.is_baryon)
+    is_composite_quark_or_lepton = property(_functions.is_composite_quark_or_lepton)
+    is_diquark = property(_functions.is_diquark)
+    is_dyon = property(_functions.is_dyon)
+    is_gauge_boson_or_higgs = property(_functions.is_gauge_boson_or_higgs)
+    is_generator_specific = property(_functions.is_generator_specific)
+    is_hadron = property(_functions.is_hadron)
+    is_lepton = property(_functions.is_lepton)
+    is_meson = property(_functions.is_meson)
+    is_nucleus = property(_functions.is_nucleus)
+    is_pentaquark = property(_functions.is_pentaquark)
+    is_quark = property(_functions.is_quark)
+    is_sm_gauge_boson_or_higgs = property(_functions.is_sm_gauge_boson_or_higgs)
+    is_special_particle = property(_functions.is_special_particle)
+    is_technicolor = property(_functions.is_technicolor)
+    is_valid = property(_functions.is_valid)
+    j_spin = property(_functions.j_spin)
+    l_spin = property(_functions.l_spin)
+    s_spin = property(_functions.s_spin)
+    three_charge = property(_functions.three_charge)
+
+
+# Verify the PDGID class has all relevant functions defined in the pdgid.functions module
 for _n in _fnames:
-    _decorator = property(getattr(_functions, _n), doc=getattr(_functions, _n).__doc__)
-    setattr(PDGID, _n, _decorator)
+    assert _n in dir(
+        PDGID
+    ), "{} missing from PDGID class! Update the list in pdgid.py".format(_n)
```

Comment:
This is much better statically, allowing "PDGID" to be checked and completed by static tools. We check these at run time and throw an error if it's not kept up (and this doesn't change that rapidly).

Reply:
Point taken.